services/checks/checks.go

```diff
@@ -166,55 +204,69 @@ func (s *Service) waitForResult(ctx context.Context, resultID string) (*models.A
 			return nil, errors.Errorf("Action %s failed: %s.", resultID, res.Error)
 		}
 
-		_, err = agentpb.UnmarshalActionQueryResult([]byte(res.Output))
+		out, err := agentpb.UnmarshalActionQueryResult([]byte(res.Output))
 		if err != nil {
 			return nil, errors.Errorf("Failed to parse action result : %s.", err)
 		}
 
-		return res, nil
+		return out, nil
 	}
 }
 
 // executeChecks runs all available checks for all reachable services.
 func (s *Service) executeChecks(ctx context.Context) {
+	s.l.Info("Executing checks...")
+
 	mySQLChecks, postgreSQLChecks, mongoDBChecks := s.groupChecksByDB(s.checks)
```

Comment:
That's why state is bad :) We are not holding a lock there.
Will fix.

Reply:
oops, yeah, checks should be accessed via `s.Checks` method